clients/src/main/java/org/apache/kafka/common/requests/ShareFetchRequest.java

```diff
@@ -226,23 +189,18 @@ public int maxWait() {
         return data.maxWaitMs();
     }
 
-    public Map<TopicIdPartition, ShareFetchRequest.SharePartitionData> shareFetchData(Map<Uuid, String> topicNames) {
+    public List<TopicIdPartition> shareFetchData(Map<Uuid, String> topicNames) {
         if (shareFetchData == null) {
             synchronized (this) {
                 if (shareFetchData == null) {
                     // Assigning the lazy-initialized `shareFetchData` in the last step
                     // to avoid other threads accessing a half-initialized object.
-                    final LinkedHashMap<TopicIdPartition, ShareFetchRequest.SharePartitionData> shareFetchDataTmp = new LinkedHashMap<>();
+                    final List<TopicIdPartition> shareFetchDataTmp = new ArrayList<>();
                     data.topics().forEach(shareFetchTopic -> {
                         String name = topicNames.get(shareFetchTopic.topicId());
                         shareFetchTopic.partitions().forEach(shareFetchPartition -> {
                             // Topic name may be null here if the topic name was unable to be resolved using the topicNames map.
-                            shareFetchDataTmp.put(new TopicIdPartition(shareFetchTopic.topicId(), new TopicPartition(name, shareFetchPartition.partitionIndex())),
-                                    new ShareFetchRequest.SharePartitionData(
-                                            shareFetchTopic.topicId(),
-                                            shareFetchPartition.partitionMaxBytes()
-                                    )
-                            );
+                            shareFetchDataTmp.add(new TopicIdPartition(shareFetchTopic.topicId(), new TopicPartition(name, shareFetchPartition.partitionIndex())));
```

Comment:
nit: This could be `new TopicIdPartition(shareFetchTopic.topicId(), shareFetchPartition.partitionIndex(), name)`.